axelrod/actions.py

```diff
@@ -9,31 +9,65 @@
 C, D = Actions.C, Actions.D
 """
 
-# Type alias for actions.
-Action = str
+from enum import Enum
+from typing import Iterable
+
+
+class UnknownActionError(ValueError):
+    def __init__(self, *args):
+        super(UnknownActionError, self).__init__(*args)
+
+
+class Actions(Enum):
+
+    C = 1
+    D = 0
+
+    def __bool__(self):
+        return bool(self.value)
 
+    def __repr__(self):
+        return '{}'.format(self.name)
 
-class Actions(object):
-    C = 'C'  # type: Action
-    D = 'D'  # type: Action
+    def __str__(self):
+        return '{}'.format(self.name)
+
+    def flip(self):
+        """Returns the opposite Action. """
+        if self == Actions.C:
+            return Actions.D
+        if self == Actions.D:
+            return Actions.C
+
+    @classmethod
+    def from_char(cls, character):
+        """Converts a single character into an Action. `Action.from_char('C')`
+        returns `Action.C`. `Action.from_char('CC')` raises an error. Use
+        `str_to_actions` instead."""
+        if character == 'C':
+            return cls.C
+        elif character == 'D':
+            return cls.D
+        else:
+            raise UnknownActionError('Character must be "C" or "D".')
```

Comment:
that's pretty close.  It originally caught the KeyError and raised a ValueError.  Any way you slice it, it has catch the KeyError and raise a new error to have a useful message, so it could be whatever error we please.  

So:

`UnknownActionError('Character must be "C" or "D".')` or `ValueError('Character must be "C" or "D".')` or `KeyError('The only accepted action keys are "C" and "D".')` 

fyi, i got curious and timed it:
10000 loops of if/else = 4ms
10000 loops of dict = 5.5ms 
so if/else saves about 150ns.

I don't have a strong feeling on this.  i kind of prefer ValueError or UnknownActionError to KeyError, and dictionary to if/else, but that's about the extent of it.  I'll happily go with "do it .... way, please." or I'll make the call if you wish.  what shall i do?